### PR TITLE
Generalize the WifiState so it can be used for other network kinds

### PR DESCRIPTION
**Pull request by kasperl**

### CODE REVIEW

File: lib/system/base/network.toit [hunk 1]
```diff
@@ -45,16 +45,67 @@ class NetworkResource extends ServiceResourceProxy:
...
+  disconnect -> none
+
+class NetworkResource extends ServiceResource:
+  state_/NetworkState ::= ?
+  constructor service/ServiceDefinition client/int .state_ --notifiable/bool=false:
+    super service client --notifiable=notifiable
+  on_closed -> none:
+    critical_do: state_.down
+
+monitor NetworkState:
```
Comment (erikcorry) [inline lib/system/base/network.toit:61]: This could use a line or two of comment to indicate what it is for.

File: lib/system/base/network.toit [hunk 2]
```diff
@@ -67,6 +118,21 @@ abstract class ProxyingNetworkServiceDefinition extends ServiceDefinition:
   */
   abstract proxy_mask -> int
 
+  /**
+  Opens the proxied network.
+
+  Subclasses may decide to use the call to establish the underlying
+    connection in which case the call may throw exceptions and
+    possibly time out. In case of such exceptions, $close_network
+    is not called.
```
Comment (erikcorry) [inline lib/system/base/network.toit:127]: Does this mean it is someone's responsibility to call close_network?  If so whose?

**Review by erikcorry — APPROVED** [review]

**Review by floitsch — APPROVED** [review]
LGTM.

File: lib/system/base/network.toit (hunk 1 above)
Comment (kasperl) [reply lib/system/base/network.toit:69]: Done.

**Pull request merged**